### PR TITLE
ci: upgrade `checkers` build to Fedora:40

### PR DESCRIPTION
**Pull request by alevenberg**
https://github.com/googleapis/google-cloud-cpp/issues/14076

In theory, the formatting is better. TBH I didn't look too hard, but it looks like all whitespace.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14138)

### CODE REVIEW

**Review by dbolduc — APPROVED** [review]

File: google/cloud/bigtable/internal/row_reader_impl.h [hunk 1]
```diff
@@ -43,7 +43,7 @@ class StatusOnlyRowReader : public RowReaderImpl {
  public:
   explicit StatusOnlyRowReader(Status s) : status_(std::move(s)) {}
 
-  void Cancel() override{};
+  void Cancel() override {};
```
Comment (coryan) [inline google/cloud/bigtable/internal/row_reader_impl.h:46]: Unrelated: the trailing semi-colon has no business here. @dbolduc FYI.

File: google/cloud/pubsub/internal/default_batch_sink.h [hunk 2]
```diff
@@ -39,7 +39,7 @@ class DefaultBatchSink : public BatchSink {
 
   ~DefaultBatchSink() override = default;
 
-  void AddMessage(pubsub::Message const&) override{};
+  void AddMessage(pubsub::Message const&) override {};
```
Comment (coryan) [inline google/cloud/pubsub/internal/default_batch_sink.h:42]: Another spurious trailing semi-colon.

File: google/cloud/pubsub/internal/sequential_batch_sink.h [hunk 3]
```diff
@@ -40,7 +40,7 @@ class SequentialBatchSink
 
   ~SequentialBatchSink() override = default;
 
-  void AddMessage(pubsub::Message const&) override{};
+  void AddMessage(pubsub::Message const&) override {};
```
Comment (coryan) [inline google/cloud/pubsub/internal/sequential_batch_sink.h:43]: Ditto.

**Review by coryan — APPROVED** [review]

**Comment by dbolduc** [comment]
Ack, let's merge this PR then deal with the semicolons.

**Comment by alevenberg** [comment]
Removed the semicolons in https://github.com/googleapis/google-cloud-cpp/pull/14140

**Pull request merged**